app/payment-link-v2/reference-confirm/reference-confirm.controller.js

```diff
@@ -33,20 +33,25 @@ function validateConfirmReferenceFormValue (confirmReference, res) {
 function getPage (req, res, next) {
   const product = req.product
 
+  const sessionReferenceNumber = paymentLinkSession.getReference(req, product.externalId)
+
   if (!product.reference_enabled) {
     return next(new NotFoundError('Attempted to access reference confirm page with a product that auto-generates references.'))
+  } else if (!sessionReferenceNumber) {
+    return next(new NotFoundError('Attempted to access reference confirm page without a reference in the session.'))
   }
 
+  const heading = res.locals.__p('paymentLinksV2.referenceConfirm.confirmYourReference').replace('%s', product.reference_label)
+
   const data = {
+    heading,
     productExternalId: product.externalId,
     productName: product.name,
     productDescription: product.description,
     paymentReferenceLabel: product.reference_label,
     paymentReferenceHint: product.reference_hint
   }
 
-  const sessionReferenceNumber = paymentLinkSession.getReference(req, product.externalId)
-
   if (sessionReferenceNumber) {
```

Comment:
As we're throwing an error if this is not present, don't need to set this on the `data` object inside this if statement, can just set it when we instantiate `data`

Reply:
Good spot - will update that